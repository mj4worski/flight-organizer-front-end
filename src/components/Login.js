import React from 'react';
import {connect} from 'react-redux';
import {tryLogin as onLoginClick} from '../actions/index';

const LoginForm = ({onLoginClick}) => {
    let _login, _password;
    const onSubmit = e => {
        e.preventDefault();
        onLoginClick(_login.value, _password.value);
        _login.value = '';
        _password.value = '';
        _login.focus();
        _password.focus();
    };

    return (
        <div>
            <p> Podaj login i aktualne hasło</p>
            <form onSubmit={onSubmit}>
                <p>Login: <input ref={input => _login = input} type="text"/></p>
                <p>Hasło : <input ref={input => _password = input} type="password"/></p>
                <button>Zaloguj</button>
            </form>
        </div>
    )
};

const Login = connect(
    null,
    {onLoginClick}
)(LoginForm);

export default Login;