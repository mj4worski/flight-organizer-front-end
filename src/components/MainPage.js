// @flow

import React from 'react';
import BestPlace from './bestplaces/BestPlace';
import './bestplaces/BestPlace.scss';

export default () => (
  <BestPlace />
);
