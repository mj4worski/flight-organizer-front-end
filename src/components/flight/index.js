// @ flow
export { default as FlightList } from './FlightList';
export { default as FindFlightsFrom } from './FindFlightsForm';
