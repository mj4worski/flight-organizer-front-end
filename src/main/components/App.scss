body {
  margin: 0;
  padding: 0;
  background: linear-gradient(to right, #ff9068, #3c2c49);
}

.app {
  display: flex;
  flex-direction: column;
  height: 100%;
}

