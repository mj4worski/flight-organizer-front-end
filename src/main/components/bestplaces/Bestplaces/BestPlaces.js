import React, { Component } from 'react';
import Place from '../PlaceTile';
import { getBestPlaces } from '../../../api';
import './BestPlaces.scss';

class BestPlaces extends Component {
  constructor(props) {
    super(props);
    this.state = {
      places: [],
    };
  }

  componentDidMount() {
    getBestPlaces().then(places => this.setState({ places }));
  }

  render() {
    const spinnerShow = this.state.places.length === 0;
    const { places } = this.state;
    return (
      <div>
        <section className="best-place">
          {
              spinnerShow ?
                <div className="best-place__spinner" /> :
                  places.map(place =>
                    (<Place
                      key={place.id}
                      {...place}
                    />))
        }
        </section>
      </div>
    );
  }
}

export default BestPlaces;
