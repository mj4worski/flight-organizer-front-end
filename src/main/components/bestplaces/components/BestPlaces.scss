.best-place {
  display: flex;
  justify-content: space-between;

  &__spinner {
    margin-top: 90px;
    border: 16px solid #fd746c;
    border-top: 16px solid #5c3fd0;
    border-radius: 50%;
    width: 50px;
    height: 50px;
    animation: spin 2s linear infinite;
  }
}

@keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}

