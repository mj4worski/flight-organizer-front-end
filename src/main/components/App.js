import React from 'react';
import {
    Router,
    Route,
} from 'react-router-dom';
import 'whatwg-fetch';
import createBrowserHistory from 'history/createBrowserHistory';
import FindFlights from './flight/FindFlightsContainer';
import MainPage from './MainPage';
import Login from './Login';
import Header from './header/Header';
import Footer from './footer/Footer';
import './App.scss';


const Content = () => (
  <div >
    <Route exact path="/" component={MainPage} />
    <Route path="/findFlight" component={FindFlights} />
    <Route path="/login" component={Login} />
  </div>
);

export default () => (
  <Router history={createBrowserHistory()}>
    <div className="app">
      <Header />
      <Content />
      <Footer />
    </div>
  </Router>
);

