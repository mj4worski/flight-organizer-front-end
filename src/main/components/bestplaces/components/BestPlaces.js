import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { fetchBestPlaces } from '../actions';
import { selectBestPlaces } from '../reducers';
import PlaceTile from '../components/PlaceTile';
import './BestPlaces.scss';

class BestPlaces extends Component {
    /* eslint-disable no-shadow */
  componentDidMount() {
    const {
        bestPlaces,
        fetchBestPlaces,
    } = this.props;
    if (bestPlaces.length === 0) {
      fetchBestPlaces();
    }
  }

  render() {
    const { bestPlaces = [] } = this.props;
    return (
      <div>
        <section className="best-place">
          {
                  bestPlaces.length === 0 ?
                    <div className="best-place__spinner" />
                    : bestPlaces.map(place => <PlaceTile key={place.id} {...place} />)
          }
        </section>
      </div>
    );
  }
}

BestPlaces.propTypes = {
  bestPlaces: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        name: PropTypes.string.isRequired,
        image: PropTypes.string,
      }),
  ),
  fetchBestPlaces: PropTypes.func.isRequired,
};

BestPlaces.defaultProps = {
  bestPlaces: [],
};

const mapStateToProps = state => ({
  bestPlaces: selectBestPlaces(state),
});

const mapDispatchToProps = {
  fetchBestPlaces,
};

export default connect(mapStateToProps, mapDispatchToProps)(BestPlaces);
export { BestPlaces };
