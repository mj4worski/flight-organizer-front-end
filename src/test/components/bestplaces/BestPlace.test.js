import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { BestPlaces } from '../../../main/components/bestplaces/components/BestPlaces';

const mockPlaceData = [{ id: 1, name: 'PlaceTile 1', image: 'Image 1' }, { id: 2, name: 'PlaceTile 2', image: 'Image 2' }];

jest.mock('../../../main/api', () => {
  const module = jest.genMockFromModule('../../../main/api');
  module.getBestPlaces = () => Promise.resolve(mockPlaceData);
  return module;
});

const fetchBestPlacesMock = jest.fn();

describe('<BestPlaces />', () => {
  it('should render <BestPlaces /> without PlaceTitle components', () => {
    const component = renderer.create(<BestPlaces fetchBestPlaces={fetchBestPlacesMock} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should calls fetchBestPlaces after render', () => {
    shallow(<BestPlaces fetchBestPlaces={fetchBestPlacesMock} />);
    expect(fetchBestPlacesMock).toHaveBeenCalled();
  });

  it('should render <BestPlaces /> with 2 PlaceTile components', () => {
    const bestPlacesData = [
      {
        id: 1,
        name: 'FirstPlace',
        image: 'SomeImage',
      },
      {
        id: 2,
        name: 'SecondPlace',
        image: 'SomeImage',
      },
    ];
    const wrapper = shallow(<BestPlaces
      fetchBestPlaces={fetchBestPlacesMock}
      bestPlaces={bestPlacesData}
    />);
    expect(wrapper.find('PlaceTile')).toHaveLength(2);
  });
});
