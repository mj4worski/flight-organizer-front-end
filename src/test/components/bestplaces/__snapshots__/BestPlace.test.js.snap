// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BestPlaces /> should render <BestPlaces /> without PlaceTitle components 1`] = `
<div>
  <section
    className="best-place"
  >
    <div
      className="best-place__spinner"
    />
  </section>
</div>
`;
